### PR TITLE
Fix all occurrences of <Section /> as well as lots of other stuff

### PR DESCRIPTION
**Pull request by Discordius**
This PR refactors all of our usages of the `Section` component, which required a large amount of refactoring in a bunch of different places as well as some significant design changes in some places. I also ended up fixing quite a few bugs and inconsistencies along the way. 

Before we merge this into master, we want to rerun the `fixDenormalizedFields` migration on conversations, since this adds a new `messageCount` field which we use to filter out empty conversations (which gets rid of the horrible user experience of clicking on the new message button and having a new conversation show up in the Inbox of the receiving user immediately).

### CODE REVIEW

File: packages/lesswrong/components/localGroups/LocalGroupPage.jsx [hunk 1]
```diff
@@ -6,114 +6,95 @@ import { Posts } from '../../lib/collections/posts';
...
     ...postBodyStyles(theme),
+    padding: theme.spacing.unit,
   }
 });
 
 class LocalGroupPage extends Component {
-  renderTitleComponent = () => {
-    const { classes } = this.props;
+  render() {
+    const { classes, document:group, currentUser } = this.props;
```
Comment (Raemon) [inline packages/lesswrong/components/localGroups/LocalGroupPage.jsx]: want to have talked about this with Jim

File: packages/lesswrong/components/localGroups/LocalGroupPage.jsx [hunk 2]
```diff
@@ -6,114 +6,95 @@ import { Posts } from '../../lib/collections/posts';
...
-        {Localgroups.options.mutations.edit.check(this.props.currentUser, group)
-          && <Components.GroupFormLink documentId={groupId} label="Edit group" />}
+      <div className={classes.root}>
+        <CommunityMapWrapper
+          terms={{view: "events", groupId: groupId}}
+          groupQueryTerms={{view: "single", groupId: groupId}}
+          mapOptions={{zoom:11, center: location, initialOpenWindows:[groupId]}}
+        />
+        <SingleColumnSection>
+          <SectionTitle title={group.name}>{currentUser && <SectionButton><SubscribeTo document={group} /></SectionButton>}</SectionTitle>
```
Comment (Raemon) [inline packages/lesswrong/components/localGroups/LocalGroupPage.jsx]: I'd personally prefer this broken up into multiple lines, ala
<SectionTitle title={group.name}>
{ blah}
</SectionTitle>

File: packages/lesswrong/components/localGroups/LocalGroupPage.jsx [hunk 3]
```diff
@@ -6,114 +6,95 @@ import { Posts } from '../../lib/collections/posts';
...
+          <div className={classes.groupDescription}>
+            <div className={classes.groupSubtitle}>
+              <SectionFooter>
+                <span className={classes.groupInfo}>
+                  <div className={classes.groupLocation}>{group.location}</div>
+                  <div className={classes.groupLinks}><GroupLinks document={group} /></div>
+                </span>
+                {Posts.options.mutations.new.check(currentUser) && 
+                  <SectionButton><Link to={{pathname:"/newPost", query: {eventForm: true, groupId}}} className={classes.leftAction}>
```
Comment (Raemon) [inline packages/lesswrong/components/localGroups/LocalGroupPage.jsx]: same here

File: packages/lesswrong/components/localGroups/LocalGroupPage.jsx [hunk 4]
```diff
@@ -6,114 +6,95 @@ import { Posts } from '../../lib/collections/posts';
...
+                  Create new event
+                </Link></SectionButton>}
+                {Posts.options.mutations.new.check(this.props.currentUser) && 
+                  <SectionButton><Link to={{pathname:"/newPost", query: {groupId}}} className={classes.leftAction}>
```
Comment (Raemon) [inline packages/lesswrong/components/localGroups/LocalGroupPage.jsx]: same

File: packages/lesswrong/components/localGroups/LocalGroupPage.jsx (hunk 1 above)
Comment (jimrandomh) [reply packages/lesswrong/components/localGroups/LocalGroupPage.jsx]: ALL BIKESHEDS MUST BE PURPLE.

**Comment by jimrandomh** [comment]
I tried hooking this up to prod and going to my PMs page, and it said I had no PMs (regardless of whether I check the Show Archived Conversations box). There's no migration script, so something's broken there.

**Comment by jimrandomh** [comment]
Fixes #1968 .

**Comment by jimrandomh** [comment]
When you receive a PM, the notifications you get in the bell-icon notifications menu and by email go to a URL that is no longer correct; it shows you your whole inbox, instead of showing you the specific conversation. You need to update URL patterns in `packages/lesswrong/lib/helpers.js` (2x) and `packages/lesswrong/lib/collections/conversations/helpers.js`
 *

**Pull request merged**